src/display_context.cpp

```diff
@@ -97,23 +97,25 @@ display_context::can_move_result display_context::unit_can_move(const unit& u) c
 	        }
 	    }
 
-		int max_distance = *std::prev(attackable_distances.end());
+		if(!attackable_distances.empty()) {
```

Comment:
Hmm, I originally thought this to be redundant (as it could only be empty if the unit has no attacks), but then I realized that it could also be empty if the unit's attacks specify a max_range lower than the min_range… which should really be an error somewhere…

Is it possible for max_range to be less than min_range? Or is that checked and sanitized elsewhere in the code (maybe in the attack constructor or something)?

Reply:
it is possible, but I think it's okay, handy for debug (manually setting values one by one), also could be used in some other way, so as long as we are prepared for that case it's going to be okay :D